File: clients/javascript/src/apis/msgs.ts
// this file is @generated

import {
    type MsgNamespaceCreateIn,
    MsgNamespaceCreateInSerializer,
} from '../models/msgNamespaceCreateIn';
import {
    type MsgNamespaceCreateOut,
    MsgNamespaceCreateOutSerializer,
} from '../models/msgNamespaceCreateOut';
import {
    type MsgNamespaceGetIn,
    MsgNamespaceGetInSerializer,
} from '../models/msgNamespaceGetIn';
import {
    type MsgNamespaceGetOut,
    MsgNamespaceGetOutSerializer,
} from '../models/msgNamespaceGetOut';
import {
    type MsgPublishIn,
    MsgPublishInSerializer,
} from '../models/msgPublishIn';
import {
    type MsgPublishOut,
    MsgPublishOutSerializer,
} from '../models/msgPublishOut';
import {
    type MsgQueueAckIn,
    MsgQueueAckInSerializer,
} from '../models/msgQueueAckIn';
import {
    type MsgQueueAckOut,
    MsgQueueAckOutSerializer,
} from '../models/msgQueueAckOut';
import {
    type MsgQueueReceiveIn,
    MsgQueueReceiveInSerializer,
} from '../models/msgQueueReceiveIn';
import {
    type MsgQueueReceiveOut,
    MsgQueueReceiveOutSerializer,
} from '../models/msgQueueReceiveOut';
import {
    type MsgStreamCommitIn,
    MsgStreamCommitInSerializer,
} from '../models/msgStreamCommitIn';
import {
    type MsgStreamCommitOut,
    MsgStreamCommitOutSerializer,
} from '../models/msgStreamCommitOut';
import {
    type MsgStreamReceiveIn,
    MsgStreamReceiveInSerializer,
} from '../models/msgStreamReceiveIn';
import {
    type MsgStreamReceiveOut,
    MsgStreamReceiveOutSerializer,
} from '../models/msgStreamReceiveOut';
import {
    type MsgTopicConfigureIn,
    MsgTopicConfigureInSerializer,
} from '../models/msgTopicConfigureIn';
import {
    type MsgTopicConfigureOut,
    MsgTopicConfigureOutSerializer,
} from '../models/msgTopicConfigureOut';
import { MsgsNamespace } from './msgsNamespace';
import { MsgsQueue } from './msgsQueue';
import { MsgsStream } from './msgsStream';
import { MsgsTopic } from './msgsTopic';
import { HttpMethod, CoyoteRequest, CoyoteRequestContext } from "../request";

export class Msgs {
    public constructor(private readonly requestCtx: CoyoteRequestContext) {}

    public get namespace() {
        return new MsgsNamespace(this.requestCtx);
    }

    public get queue() {
        return new MsgsQueue(this.requestCtx);
    }

    public get stream() {
        return new MsgsStream(this.requestCtx);
    }

    public get topic() {
        return new MsgsTopic(this.requestCtx);
    }

    /** Publishes messages to a topic within a namespace. */
        public publish(
            msgPublishIn: MsgPublishIn,
            ): Promise<MsgPublishOut> {
            const request = new CoyoteRequest(HttpMethod.POST, "/api/v1/msgs/publish");

            request.setBody(
                    MsgPublishInSerializer._toJsonObject(
                        msgPublishIn,
                    )
                );
            
                return request.send(
                    this.requestCtx,
                    MsgPublishOutSerializer._fromJsonObject,
                );
            }

        

    }

File: clients/java/src/main/java/com/svix/diom/apis/Msgs.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.ApiException;
import com.svix.diom.HttpClient;
import com.svix.diom.Utils;
import java.io.IOException;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import lombok.Getter;
import okhttp3.Headers;
import okhttp3.HttpUrl;
import com.svix.diom.models.MsgNamespaceCreateIn;
import com.svix.diom.models.MsgNamespaceCreateOut;
import com.svix.diom.models.MsgNamespaceGetIn;
import com.svix.diom.models.MsgNamespaceGetOut;
import com.svix.diom.models.MsgPublishIn;
import com.svix.diom.models.MsgPublishOut;
import com.svix.diom.models.MsgQueueAckIn;
import com.svix.diom.models.MsgQueueAckOut;
import com.svix.diom.models.MsgQueueReceiveIn;
import com.svix.diom.models.MsgQueueReceiveOut;
import com.svix.diom.models.MsgStreamCommitIn;
import com.svix.diom.models.MsgStreamCommitOut;
import com.svix.diom.models.MsgStreamReceiveIn;
import com.svix.diom.models.MsgStreamReceiveOut;
import com.svix.diom.models.MsgTopicConfigureIn;
import com.svix.diom.models.MsgTopicConfigureOut;

public class Msgs {
    private final HttpClient client;

    public Msgs(HttpClient client) {
        this.client = client;
    }

    /** Publishes messages to a topic within a namespace. */
    public MsgPublishOut publish(
        final MsgPublishIn msgPublishIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/msgs/publish");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            msgPublishIn,
            MsgPublishOut.class
            );
    }
}